cmd/snap/cmd_set.go

```diff
@@ -155,6 +161,34 @@ func (x *cmdSet) Execute([]string) error {
 	return nil
 }
 
+func validateJsonKeys(v interface{}) error {
+	var initial map[string]interface{}
+	var ok bool
+
+	if initial, ok = v.(map[string]interface{}); !ok {
+		// It's a boolean not a JSON
```

Comment:
I don't quite understand this comment.

Reply:
In this part:
https://github.com/snapcore/snapd/blob/7d1e20927714efacaf92d5c2b59393def0b39721/cmd/snap/cmd_set.go#L111-L114

There is a validation to check if the input is a valid JSON. If it passes, and then goes to the next `else` block, it must be JSON, right? **Wrong**, it can be a **boolean**. 

So, if it fails under this `if`, it means that it must be a boolean.

Not sure if this is the correct or desired behavior, but currently, if you use `-t` and pass a boolean like:
```bash
sudo snap set -t snapname key=true
```
It is accepted. So, to make this behavior consistent, I decided to return earlier in case it's a boolean.

Reply:
I think the comment is inaccurate.

There's no "JSON" type in Go apart from what is effectely a []byte - some binary with what we hope is valid JSON. The type assertion on line 168 of the diff only checks that the value given is a `map[string]any` - very much a Go type of a very specfic meaning (map from a string to an empty interface type). If you want to check that the value is a boolean please use a type switch or a type assertion that looks for the boolean value explicitly. IMO as written the comment is really confusing. Perhaps there's some contextual knowledge that makes it more true than in general but even if that is the case a direct check is better than a negative check.

Reply:
Agree! Changing to a type switch. Thanks for the comment!